### PR TITLE
HDDS-11826. Interactive mode for ozone shell

### PR DESCRIPTION
**Pull request by adoroszlai**
## What changes were proposed in this pull request?

Use JLine3 to implement an interactive shell for Ozone.  It can be started with `ozone sh --interactive`.

[demo](https://asciinema.org/a/klMulPlKQNbNIO2W6p6CdlBk3) (link valid for 7 days)

Reference:
https://github.com/remkop/picocli/tree/main/picocli-shell-jline3

https://issues.apache.org/jira/browse/HDDS-11826

## How was this patch tested?

CI:
https://github.com/adoroszlai/ozone/actions/runs/12121381248

### CODE REVIEW

**Review by nandakumar131 — APPROVED** [review]
@adoroszlai thanks a lot for working on this. This will be very useful.

Overall the changes look good to me. 
Added couple of minor nitpicks, feel free to ignore them if they don't make sense.

File: hadoop-hdds/tools/pom.xml [hunk 1]
```diff
@@ -179,20 +179,20 @@ https://maven.apache.org/xsd/maven-4.0.0.xsd">
         <artifactId>maven-compiler-plugin</artifactId>
         <configuration>
           <annotationProcessorPaths>
-            <path>
-              <groupId>org.apache.ozone</groupId>
-              <artifactId>hdds-config</artifactId>
-              <version>${hdds.version}</version>
-            </path>
```
[the remark this answers is not in the document]
Comment (adoroszlai) [reply hadoop-hdds/tools/pom.xml:186]: This is required only for processing `@Config` annotations, creating `ozone-default-generated.xml` at build time.  `hdds-tools`/`ozone-tools` do not have any `@Config` currently, so it is safe to remove.  I thought we wouldn't add CLI-only config properties, rather use CLI options for the same.

[the remark this answers is not in the document]
Comment (nandakumar131) [reply hadoop-hdds/tools/pom.xml:186]: Makes sense, thanks for the explanation!

**Comment by nandakumar131** [comment]
Thanks for the contribution @adoroszlai!

**Comment by adoroszlai** [comment]
Thanks @nandakumar131 for reviewing and merging this.

**Pull request merged**